core/src/main/scala/io/finch/Decode.scala

```diff
@@ -13,7 +12,7 @@ import shapeless.{:+:, CNil, Coproduct, Witness}
 trait Decode[A] {
   type ContentType <: String
 
-  def apply(b: Buf, cs: Charset): Try[A]
+  def apply(b: Buf, cs: Charset): Either[Throwable, A]
```

Comment:
I wonder if we can make it more specific `Either[io.finch.Error, A]`? If it's not easy, no big deal, we can revisit that later.

Reply:
I'm perfectly happy if we just do `Either[Throwable, A]` for now and revisit this later. So, please, use your own judgment.

Reply:
Okay, I will keep it `Either[Throwable, A]` for now but will look into later.